pkg/controller/deployment/deployment_controller.go

```diff
@@ -43,33 +43,39 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch for changes to primary resource StatefulSet
+	err = c.Watch(&source.Kind{Type: &appsv1.StatefulSet{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
-var _ reconcile.Reconciler = &ReconcileDeployment{}
+var _ reconcile.Reconciler = &ReconcileObject{}
 
-// ReconcileDeployment reconciles a Deployment object
-type ReconcileDeployment struct {
+// ReconcileObject reconciles a Kubernetes object
+type ReconcileObject struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
 	client client.Client
 	scheme *runtime.Scheme
 }
 
-// Reconcile reads that state of the cluster for a Deployment object and makes changes based on the state read
-// and what is in the Deployment.Spec
+// Reconcile reads that state of the cluster for a Kubernetes object and makes changes based on the
+// state read and what is in the Object Spec
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+func (r *ReconcileObject) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	log.WithFields(log.Fields{
 		"namespace": request.Namespace,
 		"name":      request.Name,
-	}).Print("Reconciling Deployment")
+	}).Print("Reconciling the Object")
 
-	// Fetch the Deployment instance
-	instance := &appsv1.Deployment{}
-	err := r.client.Get(context.Background(), request.NamespacedName, instance)
+	// Fetch the object instance
+	instance := *runtime.Object{}
```

Comment:
Can either do type checking in this function or in each of the functions in the `inject` package.

Reply:
The `Sidecar` function internally calls `decorate` which uses other parameters of the object, so it would be useful to pass the object around. Though there is #104 to consider.